### DISCUSSION TITLE
Avoiding Duplicate Password Renewal Requests in Keycloak with Microsoft Office 365 Integration

### DISCUSSION THREAD

**Discussion by matheusnascgomes** [keycloak/keycloak#34790] [Q&A]
**Title: Avoiding Duplicate Password Renewal Requests in Keycloak with Microsoft Office 365 Integration**

In our company, we use Keycloak integrated with Microsoft Office 365 as an identity provider, allowing all users to log in to the application using their Office 365 credentials. However, Office 365 has a password expiration policy that requires users to renew their passwords periodically.

During one of these password expirations, users encountered issues logging into Keycloak. In addition to needing to update their password in Office 365, Keycloak also began prompting for a password renewal, creating a double request for updating credentials and complicating the authentication process.

How can I configure Keycloak to recognize the password update directly in Office 365 and prevent a duplicate password renewal request? The goal is to redirect users to the Office identity provider for password updates upon expiration without requiring any additional action in Keycloak.

**Comment by joseaio** [comment]
Same problem here when using Google IAM (instead Office 365)
